frontend/src/component/feature/FeatureView/FeatureOverview/ManageTagsDialog/TagsInput.tsx

```diff
@@ -108,14 +108,19 @@ export const TagsInput = ({
         options: TagOption[],
         params: FilterOptionsState<TagOption>,
     ) => {
-        const filtered = filter(options, params);
+        const inputValue = params.inputValue.trim();
+
+        const filtered = filter(options, {
+            ...params,
+            inputValue,
+        });
 
-        const { inputValue } = params;
         // Suggest the creation of a new value
         const isExisting = options.some(
             (option) => inputValue === option.title,
         );
-        if (inputValue !== '' && !isExisting) {
+
+        if (inputValue.length >= 2 && !isExisting) {
```

Comment:
We're checking the filtered/trimmed value for existing here, we're certain that the trimmed/filtered value is what ends up getting created afterwards right?

Reply:
Ye. We handle that on the back end here: https://github.com/Unleash/unleash/pull/7268

But regardless, if you select an existing option from the dropdown, you get the existing value anyway.